docker/Dockerfile

```diff
@@ -216,6 +228,25 @@ RUN set -e; \
     esac; \
     truncate -s0 ~/.bash_history;
 
+RUN set -e; \
+    if [ -z $(which bazel) ]; then \
+        BAZEL_INSTALL_DIR='/usr/local/bin/'; \
+        BAZEL_VERSION='5.2.0'; \
+        BAZEL_INSTALL_PATH="$BAZEL_INSTALL_DIR/bazel-$BAZEL_VERSION"; \
+        BAZEL_ARCH=$(case "$TARGETARCH" in \
+                     'amd64') \
+                         echo 'x86_64'; \
+                         ;; \
+                     'arm64') \
+                         echo 'arm64'; \
+                         ;; \
+                   esac); \
+        sudo wget -O "$BAZEL_INSTALL_PATH" "https://github.com/bazelbuild/bazel/releases/download/$BAZEL_VERSION/bazel-$BAZEL_VERSION-linux-$BAZEL_ARCH"; \
```

Comment:
Avoid using release archive from github, it's not very accessible within GFW.

Reply:
Then where should I get it from? That is exactly where Bazel releases things, and even if there are somewhere else, a Google website will be even less accessible.

Reply:
Are we able to compile it by ourselves? Git repo is much more accessible them a http binary site.

Reply:
The installation guide suggests to compile from source, but that means more CI config for us. If you want, you may add it later or hosting a mirror somewhere within GFW or whatever you like, but I guess that should be a separate (optional and low-prio) follow-up.

Reply:
As long as it's a clearly separated section for arm, I don't care, whoever start to play with arm with figure that out with you, and it's certainly not me.

Reply:
Mostly yes.

If you look at this single command, I may (and can) work with x86_64, but if there are nothing wrong with the apt source that we added above, this is not reachable for x86_64. Even if it is reached, it is a backup for the failure cases (since we `set -e` everywhere, it is also unlikely), but not changing any existing behvaior.

I will update the arm64 images to Docker hub so if anyone wants to play, they can use them. If we want to put it to CI and want to handle cross-GFW downloading issue, we may rethink and fix it, but I don't think we will do that in the foreseeable future.

Reply:
> but if there are nothing wrong with the apt source that we added above, this is not reachable for x86_64.

That's a runtime coupling, and for whoever not familiar with existing code, also a dev time coupling. 

> Even if it is reached, it is a backup for the failure cases

We should have hard failure in this case,  not to hide it with the unintended fix.

Reply:
I would recommend have isa check rather than a which. The which should be an assertion, not condition. 
Isa check should be a oneliner, not the canonical way, because one line is mostly atomic at dev time, and the less you write, the less human error there would be.